californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java

```diff
@@ -121,7 +126,22 @@ public synchronized Endpoint getDefaultEndpoint(String uriScheme) {
 		uriScheme = uriScheme.toLowerCase();
 		Endpoint endpoint = endpoints.get(uriScheme);
 		if (null == endpoint) {
-			endpoint = createDefaultEndpoint(uriScheme);
+			if (CoAP.COAP_SECURE_URI_SCHEME.equalsIgnoreCase(uriScheme)) {
+				throw new IllegalStateException("URI scheme " + uriScheme + " requires a previous set connector!");
+			} else if (CoAP.COAP_TCP_URI_SCHEME.equalsIgnoreCase(uriScheme)) {
+				throw new IllegalStateException("URI scheme " + uriScheme + " requires a previous set connector!");
+			} else if (CoAP.COAP_SECURE_TCP_URI_SCHEME.equalsIgnoreCase(uriScheme)) {
+				throw new IllegalStateException("URI scheme " + uriScheme + " requires a previous set connector!");
+			} else {
+				endpoint = new CoapEndpoint();
+			}
+			try {
+				endpoint.start();
+				LOGGER.info("created implicit endpoint {} for {}", new Object[] { endpoint.getUri(), uriScheme });
+			} catch (IOException e) {
+				LOGGER.error("could not create {} endpoint", uriScheme, e);
```

Comment:
shouldn't we throw `IllegalStateException` here as well? It looks like we are putting the *unstartable* endpoint to the map so that it will be returned to clients on subsequent invocations ...

Reply:
Though the factory-nature is "gone" in the meantime, I received irritating NullPointerExceptions later. But I didn't receive such a IOException. Therefore I created this PR to indicate the cause of actual errors and didn't touch the other stuff.

Reply:
@sophokles73 
So, should we really encapsulate the IOException (which never seems to occur), or can we go with this PR "as it is"?